docs/metrics.md

```diff
@@ -0,0 +1,161 @@
+
+### List of Metrics Exported By Object Storage
+- Thanos leverages object storage solutions such as Minio, Amazon S3, Google Cloud Storage, etc.
+
+  | Metric Name                                        | Type      | Description                                                                                                                                                          |
+  |----------------------------------------------------|-----------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+  | thanos_objstore_bucket_last_successful_upload_time | gauge     | Second timestamp of the last successful upload to the bucket.                                                                                                        |
+  | thanos_objstore_bucket_operation_duration_seconds  | histogram | Duration of successful operations against the bucket.                                                                                                                |
+  | thanos_objstore_bucket_operation_failures_total    | counter   | Total number of operations against a bucket that failed, but were not expected to fail in certain way from caller perspective. Those errors have to be investigated. |
+  | thanos_objstore_bucket_operations_total            | counter   | Total number of all attempted operations against a bucket.                                                                                                           |
+
+### List of Metrics Exported By Thanos Replicator
+- The Thanos Replicator manages the replication of Thanos block data across different object storage systems.
+
+  | Metric Name                                       | Type      | Description                                                     |
+  |---------------------------------------------------|-----------|-----------------------------------------------------------------|
+  | thanos_replicate_blocks_already_replicated_total  | counter   | Total number of blocks skipped due to already being replicated. |
+  | thanos_replicate_blocks_replicated_total          | counter   | Total number of blocks replicated.                              |
+  | thanos_replicate_objects_replicated_total         | counter   | Total number of objects replicated.                             |
+  | thanos_replicate_replication_runs_total           | counter   | The number of replication runs split by success and error.      |
+  | thanos_replicate_replication_run_duration_seconds | histogram | The Duration of replication runs split by success and error.    |
+
+### List of Metrics Exported By Thanos Store Gateway Cache
```

Comment:
If these are all metrics of the Store Gateway, they should be documented in the Store Gateway file, imho.

Reply:
I will move the Store Gateway Cache metrics to the Store Gateway PR accordingly.